src/app/_hooks/useSearch.ts

```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react'
+
+import { NextServerSearchParams } from '@/types/searchParams'
+
+import { normalizeQueryParam } from '@/utils/queryUtils'
+
+import { SEARCH_KEY } from '@/constants/searchParams'
+
+type UseSearchProps<Entry extends Record<string, unknown>> = {
+  searchParams: NextServerSearchParams
+  entries: Entry[]
+  searchBy: keyof Entry | Array<keyof Entry>
+}
+
+function search<Entry extends Record<string, unknown>>(
```

Comment:
Would this function be more explicit as `matchesQuery` or something like that?

Reply:
Yes, it would :)